sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4456,6 +4456,24 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         """.stripMargin),
       Seq(Row(2), Row(1)))
   }
+
+  test("SPARK-39444: Add OptimizeSubqueries into nonExcludableRules list") {
+    withSQLConf(
+      SQLConf.OPTIMIZER_EXCLUDED_RULES.key ->
+        "org.apache.spark.sql.catalyst.optimizer.Optimizer$OptimizeSubqueries") {
```

Comment:
nit but can we use a class reference instead of a string?

Reply:
and let's match the test place with https://github.com/apache/spark/pull/36847

Reply:
+1 for @HyukjinKwon 's comment.

Reply:
fixed.